x-pack/platform/plugins/shared/inference/common/utils/combine_signal.ts

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+export function combineSignal(left?: AbortSignal, right?: AbortSignal) {
+  if (!right) {
+    return left;
+  }
+
+  const controller = new AbortController();
+
+  left?.addEventListener('abort', () => {
+    controller.abort();
+  });
+
+  right?.addEventListener('abort', () => {
+    controller.abort();
+  });
```

Comment:
Do we need to add `{ once: true }` here? So the listener is automatically removed after it's called?

Reply:
I think this also will automatically get gc'ed